skops/card/tests/test_card.py

```diff
@@ -300,7 +311,100 @@ def test_with_metadata(self, card: Card, meth):
+            "  search_results=Table(3x2),\n"
             ")"
         )
         result = meth(card)
         assert result == expected
+
+
+class TestPlotSection:
+    def test_format_path_is_str(self):
+        section = PlotSection(alt_text="some title", path="path/plot.png")
+        expected = "![some title](path/plot.png)"
+        assert section.format() == expected
+
+    def test_format_path_is_pathlib(self):
+        section = PlotSection(alt_text="some title", path=Path("path") / "plot.png")
+        expected = f"![some title](path{os.path.sep}plot.png)"
+        assert section.format() == expected
+
+    @pytest.mark.parametrize("meth", [str, repr])
+    def test_str_and_repr(self, meth):
+        section = PlotSection(alt_text="some title", path="path/plot.png")
+        expected = "'path/plot.png'"
+        assert meth(section) == expected
+
+    def test_str(self):
+        section = PlotSection(alt_text="some title", path="path/plot.png")
+        expected = "'path/plot.png'"
+        assert str(section) == expected
+
+
+class TestTableSection:
+    @pytest.fixture
+    def table_dict(self):
+        return {"split": [1, 2, 3], "score": [4, 5, 6]}
+
+    def test_table_is_dict(self, table_dict):
+        section = TableSection(table=table_dict)
+        expected = """|   split |   score |
+|---------|---------|
+|       1 |       4 |
+|       2 |       5 |
+|       3 |       6 |"""
+        assert section.format() == expected
+
+    def test_table_is_dataframe(self, table_dict):
+        pd = pytest.importorskip("pandas")
+        df = pd.DataFrame(table_dict)
+        section = TableSection(table=df)
+        expected = """|   split |   score |
+|---------|---------|
+|       1 |       4 |
+|       2 |       5 |
+|       3 |       6 |"""
+        assert section.format() == expected
+
+    @pytest.mark.parametrize("meth", [str, repr])
+    def test_str_and_repr_table_is_dict(self, table_dict, meth):
+        section = TableSection(table=table_dict)
+        expected = "Table(3x2)"
+        assert meth(section) == expected
+
+    @pytest.mark.parametrize("meth", [str, repr])
+    def test_str_and_repr_table_is_dataframe(self, table_dict, meth):
+        pd = pytest.importorskip("pandas")
+        df = pd.DataFrame(table_dict)
+        section = TableSection(table=df)
+        expected = "Table(3x2)"
+        assert meth(section) == expected
+
+    def test_raise_error_if_no_cols(self):
+        msg = "Empty table added"
+        with pytest.raises(ValueError, match=msg):
+            TableSection(table={})
+
+    def test_raise_error_if_no_rows(self):
+        msg = "Empty table added"
+        with pytest.raises(ValueError, match=msg):
+            TableSection(table={"col": []})
+
+    def test_raise_error_df_empty(self):
+        pd = pytest.importorskip("pandas")
+        df = pd.DataFrame([])
+        msg = "Empty table added"
+        with pytest.raises(ValueError, match=msg):
+            TableSection(table=df)
```

Comment:
these could be a single parameterized test.

Reply:
Done. Please review again.